src/storage/src/source/reclock.rs

```diff
@@ -254,6 +249,10 @@ where
         // the reclock operators fully general.
         let mut into_times = MutableAntichain::new();
 
+        let mut minimum = IntoTime::minimum();
+        minimum.advance_by(inner.since.frontier());
+        into_times.update_iter([(minimum, 1)]);
```

Comment:
This is only required if you reclock `C` in your big example, right?

Reply:
`C` but also `A` and `B` and in general it's needed to reclock all timestamps that are not beyond the since

Reply:
How did it work in the past?

Reply:
It didn't, we were returning an error earlier in this function (`NotBeyondSince`)